zerver/tests/test_message_fetch.py

```diff
@@ -1826,10 +1857,107 @@ def test_get_visible_messages_with_narrow_dm(self) -> None:
         narrow = [dict(operator="dm", operand=self.example_user("iago").email)]
         self.message_visibility_test(narrow, message_ids, 2)
 
+    def test_get_messages_with_narrow_dm_including(self) -> None:
+        """
+        A request for old messages with a narrow by "dm-including" only
+        returns DMs (both group and one-on-one) with that user.
+        """
+        me = self.example_user("hamlet")
+
+        iago = self.example_user("iago")
+        cordelia = self.example_user("cordelia")
+        othello = self.example_user("othello")
+
+        matching_message_ids = []
+
+        matching_message_ids.append(
+            self.send_huddle_message(
+                me,
+                [iago, cordelia, othello],
+            ),
+        )
+
+        matching_message_ids.append(
+            self.send_huddle_message(
+                me,
+                [cordelia, othello],
+            ),
+        )
```

Comment:
It would be good to include a message to a huddle from someone other than `me`; only including messages sent by yourself to a huddle is a class of bug I can imagine in this code path.

Reply:
Expanded these test cases for other senders of huddle messages.